Gems/Atom/Feature/Common/Assets/Materials/Types/Debug/DebugVertexStreams.azsl

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <viewsrg.srgi>
+#define UvSetCount 2
+#include <Atom/RPI/ShaderResourceGroups/DefaultObjectSrg.azsli>
+#include <Atom/RPI/ShaderResourceGroups/DefaultDrawSrg.azsli>
+#include <Atom/RPI/TangentSpace.azsli>
+
+
+option enum class DebugVertexStream { Normals, Tangents, Bitangents, UVs, TangentW } o_debugVertexStream = DebugVertexStream::Normals;
+option enum class ReconstructionMethods { UseVertexData, UseSurfaceGradient, ReconstructBitangent} o_reconstructionMethod = ReconstructionMethods::UseVertexData;
+
+struct VSInput
+{
+    // Base fields (required by the template azsli file)...
+    float3 m_position : POSITION;
+    float3 m_normal : NORMAL;
+    float4 m_tangent : TANGENT; 
+    float3 m_bitangent : BITANGENT; 
+ 
+    // Extended fields (only referenced in this azsl file)...
+    float2 m_uv0 : UV0;
+};
+
+struct VSOutput
+{
+    // Base fields (required by the template azsli file)...
+    // "centroid" is needed for SV_Depth to compile
+    precise linear centroid float4 m_position : SV_Position;
+    float3 m_normal: NORMAL;
+    float4 m_tangent : TANGENT; 
+    float3 m_bitangent : BITANGENT; 
+    float3 m_worldPosition : UV0;
+
+    // VertexHelper expects the output tangent to be stripped of the w component, so we preserve it here
+    float m_bitangentSign : UV8;
+
+    // Extended fields (only referenced in this azsl file)...
+    float2 m_uv[UvSetCount] : UV1;
+};
+
+VSOutput MainVS(VSInput IN)
+{
+    VSOutput OUT;    
+    OUT.m_worldPosition = mul(ObjectSrg::GetWorldMatrix(), float4(IN.m_position, 1.0)).xyz;
+    OUT.m_position = mul(ViewSrg::m_viewProjectionMatrix, float4(OUT.m_worldPosition, 1.0));
+
+    // Only UV0 is supported
+    OUT.m_uv[0] = IN.m_uv0;
+    OUT.m_uv[1] = IN.m_uv0;    
+    
+    float4x4 objectToWorld = ObjectSrg::GetWorldMatrix();
+    float3x3 objectToWorldIT = ObjectSrg::GetWorldMatrixInverseTranspose();
+
+    ConstructTBN(IN.m_normal, IN.m_tangent, IN.m_bitangent, objectToWorld, objectToWorldIT, OUT.m_normal, OUT.m_tangent.xyz, OUT.m_bitangent);
+
+    OUT.m_tangent.w = IN.m_tangent.w;
+    return OUT;
+}
+
+struct PixelOutput
+{
+    float4 m_color : SV_Target0;
+};
+
+// Represent a vector in the (-1, -1, -1) to (1, 1, 1) range as a color in the (0, 0, 0) to (1, 1, 1) range
+// Color key
+// + x-axis: Light Coral
+// - x-axis: Teal
+// + y-axis: Bright Green
+// - y-axis: Dark Magenta
+// + z-axis: Medium Slate Blue
+// - z-axis: Olive
+float3 OffsetColor(float3 color)
```

Comment:
Here I suggest having a choice between two display modes:
1. The existing with space transformed to [0..1]
2. The original one with space between [-1..1] and the reason is that sometimes it's better to look at half space (the positive) to get a feel of the result than looking at a shifted field where middle point translated in 0.5 - in many times it brings better understanding about how half space is behaving.

Reply:
Done.